subsys/testsuite/include/zephyr/interrupt_util.h

```diff
@@ -197,6 +197,14 @@ static inline void trigger_irq(int irq)
 	z_mips_enter_irq(irq);
 }
 
+#elif defined(CONFIG_VIM)
+extern void z_soc_arm_enter_irq(int);
+
+static inline void trigger_irq(int irq)
+{
+	z_soc_arm_enter_irq(irq);
```

Comment:
where is this defined?

Reply:
That should be defined by the soc files or should call z_vim_arm_enter_irq?

Reply:
well no idea, it depends on the controller. `trigger_irq` should be able to trigger an IRQ in software, it is used for tests to software-trigger IRQs.

Reply:
The SoC which is enabling CONFIG_VIM should define z_soc_arm_enter_irq (which inturn calls z_vim_arm_enter_irq which is defined in intc_vim.c?

Reply:
again, it depends. First of all you are doing on `#elif` on symbols related to architectures with a symbol related to an interrupt controller so I don't think this is ever going to work (did you test this patch?). Second, that depends on the architecture, some arch can trigger a software IRQ by probing interrupt controller registers, others in a different way. So, it depends on your architecture I'm not familiar with.

Reply:
Yes, I tested the patch and the twister also did passed/skipped all the test cases.
In VIM, software interrupts can be triggered by configuring the interrupt controller registers.
I will check once with @vaishnavachath

Reply:
Yes @slpp95prashanth for VIM, interrupts can be triggered from SW through the VIM_RAW_* registers , J721E TRM R5FSS Registers->R5FSS_VIM Registers:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/37137606/f91b16ff-5293-4e42-a8ce-aa5b8f7d29a4)

@carlocaione there seems to be symbols specific to interrupt controllers as well in the file like `CONFIG_GIC` , what should be the acceptable approach for implementing the testsuite in this case?

Reply:
I'm fine with `ifdef`-fing on `CONFIG_VIM` but guarded by the architecture `ifdef`, pretty much like it is done for `CONFIG_GIC`.

Reply:
@carlocaione trigger_irq for gic guarded by CONFIG_GIC is a #elif directive not under architecture in subsys/testsuite/include/zephyr/interrupt_util.h. Please correct me if I am wrong.

Reply:
Yes, because GIC in this case can be used by several architectures but in your case it should be ok doing something:
```
#ifdef ARCHITECTURE
#ifdef CONFIG_VIM
....
```